packages/util/src/signature.ts

```diff
@@ -103,41 +133,47 @@ export const toCompactSig = function (v: bigint, r: Buffer, s: Buffer, chainId?:
 
 /**
  * Convert signature format of the `eth_sign` RPC method to signature parameters
- *
  * NOTE: For an extracted `v` value < 27 (see Geth bug https://github.com/ethereum/go-ethereum/issues/2053)
  * `v + 27` is returned for the `v` value
  * NOTE: After EIP1559, `v` could be `0` or `1` but this function assumes
  * it's a signed message (EIP-191 or EIP-712) adding `27` at the end. Remove if needed.
  */
-export const fromRpcSig = function (sig: string): ECDSASignature {
+export const fromRpcSig = function (sig: string, chainId?: bigint): ECDSASignature {
```

Comment:
Why do we need to add the `chainId` here? I think `calculateSigRecovery()` should also work without, so it shouldn't be necessary to change the function signature?